sympy/ntheory/factor_.py

```diff
@@ -1309,16 +1317,15 @@ def factorint(n, limit=None, use_trial=True, use_rho=True, use_pm1=True,
         else:
             print('Factoring', n)
 
-    if use_trial:
-        # this is the preliminary factorization for small factors
-        small = 2**15
-        fail_max = 600
-        small = min(small, limit or small)
-        if verbose:
-            print(trial_int_msg % (2, small, fail_max))
-        n, next_p = _factorint_small(factors, n, small, fail_max)
-    else:
-        next_p = 2
+    # this is the preliminary factorization for small factors
+    # We want to guarantee that there are no small prime factors,
+    # so we run even if `use_trial` is False.
```

Comment:
Why penalize every routine that knows it doesn't want small factorization? The flag defaults to True. If a routine wants to set it to false, it should be able to do so...and accept the consequences.

Reply:
The cost of this process is negligible. It would be more beneficial to guarantee that there are no small prime factors.

With `use_trial=False`, `next_p` is always 2. We can assume that a user who sets `use_trial=False` and uses it already knows that there are no small prime factors and uses `factorint`, but `factorint` cannot guarantee it.

The absence of small prime factors is very important information in prime factorization. We want `factorint` and its subroutines to actively use this information. For example, when `_check_termination` is executed, `_perfect_power` is called, and if `next_p` is small, trial factoring is performed each time. Therefore, it is more efficient if `next_p` is of a certain size.

This change does not mean that trial factoring is performed to find prime factors (which the user does not want). It is done so that `factorint` can compute them efficiently.

Reply:
ok